lib/client/doc.js

```diff
@@ -1035,17 +1037,46 @@ Doc.prototype._hardRollback = function(err) {
 
   // Fetch the latest version from the server to get us back into a working state
   var doc = this;
-  this._fetch({force: true}, function() {
+  this._fetch({force: true}, function(fetchError) {
     // We want to check that no errors are swallowed, so we check that:
     // - there are callbacks to call, and
     // - that every single pending op called a callback
     // If there are no ops queued, or one of them didn't handle the error,
     // then we emit the error.
+
+    if (fetchError) {
+      // This is critical error as it means that our doc is not in usable state
+      // anymore, we should throw doc error.
+      logger.error('Hard rollback doc fetch failed.', fetchError, inflightOp);
+      doc.emit('error', fetchError);
```

Comment:
Let's create a new error and concat the `fetchError` message on, e.g. "Hard rollback doc fetch error: _____"